proxy/reverse.go

```diff
@@ -203,3 +210,42 @@ func maybeSetForwardedFor(req *http.Request) {
 	}
 	req.Header.Set("X-Forwarded-For", clientIP)
 }
+
+const proxyLoopHeaderKey = "X-Etcd-Proxy-Hop-Seen-By"
+
+// recordProxy records proxyID and an endpoint in HTTP headers,
+// in order to detect a proxy loop.
+func recordProxy(req *http.Request, proxyID string, loc url.URL) {
+	pv := proxyID + "=" + loc.String()
+	if prior, ok := req.Header[proxyLoopHeaderKey]; ok {
+		pv = strings.Join(prior, ", ") + ", " + pv
```

Comment:
i would like `strings.Join(append(prior, pv), ", ")` because you don't need to write `", "` twice.